src/sdk/pynni/nni/compression/torch/knowledge_distill.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import torch
+import torch.nn.functional as F
+
+_logger = logging.getLogger(__name__)
+
+
+class KnowledgeDistill():
+    """
+    Knowledge Distill support while fine-tuning the compressed model
+    Geoffrey Hinton, Oriol Vinyals, Jeff Dean
+    "Distilling the Knowledge in a Neural Network"
+    https://arxiv.org/abs/1503.02531
+    """
+
+    def __init__(self, teacher_model, kd_T, kd_beta):
+        """
+        Parameters
+        ----------
+        teacher_model : pytorch model
+            the teacher_model for teaching the student model, it should be pretrained
+        kd_T: float
+            kd_T is the temperature parameter, when kd_T=1 we get the standard softmax function
+            As kd_T grows, the probability distribution generated by the softmax function becomes softer
+        kd_beta: float
+            kd_beta is the weight of distillation loss relative to the student loss
+        """
+
+        self.teacher_model = teacher_model
+        self.kd_T = kd_T
+        self.kd_beta = kd_beta
+
+    def get_kd_loss(self, data, student_out):
+        """
+        Parameters
+        ----------
+        data : torch.Tensor
+            the input training data
+        student_out: torch.Tensor
+            output of the student network
+
+        Returns
+        -------
+        torch.Tensor
+            weighted distillation loss
+        """
+
+        with torch.no_grad():
+            kd_out = self.teacher_model(data)
+        soft_log_out = F.log_softmax(student_out / self.kd_T, dim=1)
+        soft_t = F.softmax(kd_out / self.kd_T, dim=1)
```

Comment:
It is very common that the output of a model is not  a tensor.  Here we require the model output to be a tensor, it is a limitation.

Reply:
If the output of the neural network is not a tensor, how can the loss and gradients be back-propagated?

Reply:
the output can be a tuple or a list of tensors, such as object detection, or any multitask model.

Reply:
Thanks for the suggestion, I'll do some investigations

Reply:
I made some modifications to enable users to modify the teacher model's output(such as take the first tensor in a tuple), in this way they can decide which part to distill.